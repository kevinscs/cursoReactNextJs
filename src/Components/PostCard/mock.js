export const postCardPropsMock = {
    title: 'Titulo 1',
    body: 'Corpo 1',
    id: 1,
    cover: 'img/img.png'
}