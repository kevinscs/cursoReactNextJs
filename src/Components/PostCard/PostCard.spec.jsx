import { render, screen } from '@testing-library/react';
import { PostCard } from '.';
import { postCardPropsMock } from './mock';

const props = postCardPropsMock;

describe('<PostCard />', () => {
    it('Should render PostCard correctly', () => {
        render(<PostCard {...props} />);

       expect(screen.getByRole('img', { name: 'Titulo 1' }))
       .toHaveAttribute('src', 'img/img.png');
       expect(screen.getByRole('heading', { name: 'Titulo 1 1' })).toBeInTheDocument();
       expect(screen.getByText('Corpo 1')).toBeInTheDocument();
    });
    it('Should match snapshot', () => {
        const { container } = render(<PostCard {...props} />);
        expect(container.firstChild).toMatchSnapshot();
    })
});
