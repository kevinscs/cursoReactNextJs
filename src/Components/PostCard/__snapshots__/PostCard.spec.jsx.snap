// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> Should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="Titulo 1"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h2>
      Titulo 1
       
      1
    </h2>
    <p>
      Corpo 1
    </p>
  </div>
</div>
`;
